ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/ClientImpl.java

```diff
@@ -213,6 +246,48 @@ private HttpClientRequest configureBasicAuth(final HttpClientRequest request) {
     }
   }
 
+  private static void handleInsertIntoResponse(
+      final HttpClientResponse response,
+      final CompletableFuture<Void> cf,
+      final int numRows
+  ) {
+    if (response.statusCode() == OK.code()) {
+      response.bodyHandler(buffer -> {
```

Comment:
I see that you're waiting for the whole body and parsing it manually here.
When you come to do the streaming insert response this technique won't work and you'll need to use RecordParser - and then you'll end up with two ways of parsing the response.
I think it would be simpler and easier to just use RecordParser for all cases - this will just spit out the response lines so you don't have to do any manual parsing.

Reply:
If we don't support batch inserts, then this becomes a lot simpler too :)

Reply:
Switched to RecordParser. Might have to refactor this again when we support streaming inserts, though. We'll see.